signer/migrations/0011__materialized_chains.sql

```diff
@@ -0,0 +1,148 @@
+-- This table holds the canonical chain information for Bitcoin and Stacks
+-- chains keyed by the bitcoin chain tip.
+-- 
+-- This table includes one row for each Stacks block which is anchored to a
+-- Bitcoin block in the canonical chain. The `bitcoin_chain_tip` column is used
+-- to identify the specific chain tip for which the canonical chain is
+-- materialized. If no Stacks block(s) exist for a given Bitcoin block, the
+-- `stacks_block_hash` and `stacks_block_height` columns will be `NULL` for that
+-- Bitcoin block hash.
+CREATE TABLE sbtc_signer.canonical_chains (
+    run_id INT NOT NULL,
+    bitcoin_chain_tip BYTEA NOT NULL,
+    bitcoin_block_hash BYTEA NOT NULL,
+    bitcoin_block_height BIGINT NOT NULL,
+    stacks_block_hash BYTEA,
+    stacks_block_height BIGINT
+);
+
+-- An intermediate view to `canonical_chains` which can be used in queries to
+-- avoid needing to change query syntax if the underlying data source changes.
+-- Note that this view will return multiple canonical chains, so you need to
+-- filter the results to the specific chain tip which you are interested in.
+CREATE VIEW sbtc_signer.canonical_chains_view AS
+SELECT
+    bitcoin_chain_tip,
+    bitcoin_block_hash,
+    bitcoin_block_height,
+    stacks_block_hash,
+    stacks_block_height
+FROM sbtc_signer.canonical_chains;
```

Comment:
I think I would forgo this views. It seems like very thin layers on top of the `canonical_chains` table, so I would always query that directly.

Reply:
After [545214b](https://github.com/stacks-network/sbtc/pull/1171/commits/545214b76c2d53b3ff73a6c88cee274c2ad16eea) you'll see why I had this here :)